The Straits Times.js

```diff
@@ -0,0 +1,286 @@
+{
+	"translatorID": "2de01dfe-9572-4775-bf7e-6b55c95d60b0",
+	"label": "The Straits Times",
+	"creator": "Robert Sim",
+	"target": "^https?:\\/\\/(www.)?straitstimes.com\\/",
+	"minVersion": "3.0",
+	"maxVersion": "",
+	"priority": 100,
+	"inRepository": true,
+	"translatorType": 4,
+	"browserSupport": "gcsibv",
+	"lastUpdated": "2020-12-29 09:10:26"
+}
+
+/*
+	***** BEGIN LICENSE BLOCK *****
+
+	Copyright © 2020 Robert Sim
+
+	This file is part of Zotero.
+
+	Zotero is free software: you can redistribute it and/or modify
+	it under the terms of the GNU Affero General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	Zotero is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+	GNU Affero General Public License for more details.
+
+	You should have received a copy of the GNU Affero General Public License
+	along with Zotero. If not, see <http://www.gnu.org/licenses/>.
+
+	***** END LICENSE BLOCK *****
+*/
+
+function detectWeb(doc, url) {
+	if (url.match(/(advertise-with-the-straits-times|newsletter-signup)/)) {
```

Comment:
It feels like there must be a better heuristic than this to exclude article-like non-articles. The "Advertise" page has the URL `https://www.straitstimes.com/advertise-with-the-straits-times`, while a normal article takes the form `https://www.straitstimes.com/asia/east-asia/asean-china-foreign-ministers-meet-discuss-trade-covid-19-myanmar` - could you match on the number of path segments, or are there exceptions?

Reply:
updated